packages/experimental/src/inbox-note/inbox-dismiss-confirmation-modal.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+type ConfirmationModalProps = {
+	onClose: () => void;
+	onDismiss: () => void;
+	buttonLabel?: string;
+};
+
+const InboxDismissConfirmationModal: React.FC< ConfirmationModalProps > = ( {
+	onClose,
+	onDismiss,
+	buttonLabel = __( "Yes, I'm sure", 'woocommerce-admin' ),
+} ) => {
+	const [ inAction, setInAction ] = useState( false );
+
+	return (
+		<Modal
+			title={ __( 'Are you sure?', 'woocommerce-admin' ) }
+			onRequestClose={ () => onClose() }
+			className="woocommerce-inbox-dismiss-confirmation_modal"
+		>
+			<div className="woocommerce-inbox-dismiss-confirmation_wrapper">
+				<p>
+					{ __(
+						'Dismissed messages cannot be viewed again',
+						'woocommerce-admin'
+					) }
+				</p>
+				<div className="woocommerce-inbox-dismiss-confirmation_buttons">
+					<Button isSecondary onClick={ () => onClose() }>
+						{ __( 'Cancel', 'woocommerce-admin' ) }
+					</Button>
+					<Button
+						isSecondary
+						isBusy={ inAction }
+						disabled={ inAction }
+						onClick={ () => {
+							setInAction( true );
+							onDismiss();
+						} }
+					>
+						{ buttonLabel }
+					</Button>
+				</div>
+			</div>
+		</Modal>
+	);
+};
+
+export { InboxDismissConfirmationModal };
```

Comment:
Nit pick but any reason why you don't just export it on the line its declared?

Reply:
No particular reason, will export it directly